File: civitasRuby/pkg/civitasRuby-0.0.1/lib/civitas/sorpresa_ir_carcel.rb
#encoding:utf-8
=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas

class SorpresaIrCarcel < Sorpresa
  
  def initialize (tab)
    super("*Ir carcel*", tab, -1, "", nil)
  end
  
  def aplicarAJugador (actual, todos)
    if jugadorCorrecto(actual, todos)
      informe(actual, todos)
      todos[actual].encarcelar(@tablero.numCasillaCarcel)
    end
  end
  
end

end

File: civitasJava/src/civitas/SorpresaIrCarcel.java

package civitas;

import java.util.ArrayList;

public class SorpresaIrCarcel extends Sorpresa {
    
    SorpresaIrCarcel (Tablero tab) {
        super("*Ir carcel*");
        tablero = tab;
        texto = " Vas a la carcel... ";
    }
    
    /** Se encarcela la jugador.  */
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe (actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    
}
